### PR TITLE
Issue 6874 - 2.7.0-RC Auth doesn't use the new Flash component

### PR DESCRIPTION
**Pull request by highstrike**
#6874

Replaced "Flash." to "Message." for backwards compatibility.
Ported the default element functionality from `Session->flash` into `Flash->render`

### CODE REVIEW

File: lib/Cake/View/Helper/FlashHelper.php [hunk 1]
```diff
@@ -83,7 +83,16 @@ public function render($key = 'flash', $options = array()) {
 		}
 
 		$flash = $options + $flash;
-		CakeSession::delete("Flash.$key");
+		CakeSession::delete("Message.$key");
+
+		// backwards compatibility with Session->setFlash()
+		if($flash['element'] === 'default') {
```
Comment (tanuck) [inline lib/Cake/View/Helper/FlashHelper.php]: PHPCS is failing on this line.

Comment (dereuromark) [inline lib/Cake/View/Helper/FlashHelper.php]: Yeah, needs a space after `if`

Comment (Highstrike) [inline lib/Cake/View/Helper/FlashHelper.php]: Fixed

**Comment by dereuromark** [comment]
I don't quite get why you need to adjust the helper. Shouldn't adjusting the AuthComponent itself suffice (in order to store the flash message just as it used to in 2.6)?

**Comment by Highstrike** [comment]
@dereuromark because of backwards compatibility. If i adjust only AuthComponent to use the Flash component it will throw this notice `Element Not Found: Elements\Flash\default.ctp`. That functionality is present [here](https://github.com/cakephp/cakephp/blob/2.7/lib/Cake/View/Helper/SessionHelper.php#L144)

**Comment by dereuromark** [comment]
Sounds like someone forgot to add the default skeleton templates @ https://github.com/cakephp/cakephp/tree/2.7/lib/Cake/Console/Templates/skel/View/Elements/

**Comment by Highstrike** [comment]
@dereuromark is this better?

**Comment by markstory** [comment]
Looks reasonable, making the FlashComponent more backwards compatible is a good idea for 2.x :+1:

**Comment by markstory** [comment]
There are [more failing tests](https://travis-ci.org/cakephp/cakephp/jobs/68151080).

**Comment by Highstrike** [comment]
I've fixed the failures but I'm not sure how to fix the errors because of backwards compatibility...
```
<div id="notificationLayout">
	<h1><?php echo $params['name']; ?></h1>
	<h3><?php echo $params['title']; ?></h3>
	<p><?php echo $message; ?></p>
</div>
```
I see this is the default flash message now, however 2.6 has [another default](https://github.com/cakephp/cakephp/blob/2.6/lib/Cake/View/Helper/SessionHelper.php#L136)

- "Message" instead of "Layout"
- Class is skipped entirely
- H1, H3 and p added

Please advise.

**Comment by markstory** [comment]
If the HTML changes a bit I think that's ok. I can take a look at the issue as from the test runs I'm not clear on what the problem is.

**Comment by Highstrike** [comment]
Thanks Mark.

**Comment by markstory** [comment]
Tests fixed and merged in cbd45f702d410c90d1bc1cccca8bbc906aaf807f.

**Pull request merged**